controllers/controllers/services/instances/managed/controller.go

```diff
@@ -261,16 +276,88 @@ func (r *Reconciler) processProvisionOperation(
 			Reason:             "ProvisionFailed",
 			Message:            lastOpResponse.Description,
 		})
+		setObservedGeneration(serviceInstance)
 		return ctrl.Result{}, k8s.NewNotReadyError().WithReason("ProvisionFailed")
 	}
 
 	return ctrl.Result{}, k8s.NewNotReadyError().WithReason("ProvisionInProgress").WithRequeue()
 }
 
+func (r *Reconciler) updateServiceInstance(
+	ctx context.Context,
+	serviceInstance *korifiv1alpha1.CFServiceInstance,
+	assets osbapi.ServiceInstanceAssets,
+	osbapiClient osbapi.BrokerClient,
+) (osbapi.UpdateResponse, error) {
+	log := logr.FromContextOrDiscard(ctx).WithName("update-service-instance")
+
+	serviceInstance.Status.LastOperation = korifiv1alpha1.LastOperation{
+		Type:  "update",
+		State: "in progress",
+	}
+
+	var updateResponse osbapi.UpdateResponse
+	var err error
+	updateResponse, err = osbapiClient.Update(ctx, osbapi.UpdatePayload{
+		InstanceID: serviceInstance.Name,
+		UpdateRequest: osbapi.UpdateRequest{
+			ServiceId: assets.ServiceOffering.Spec.BrokerCatalog.ID,
+			PlanID:    assets.ServicePlan.Spec.BrokerCatalog.ID,
+		},
+	})
+	if err != nil {
+		log.Error(err, "failed to update service")
+
+		if osbapi.IsUnrecoveralbeError(err) {
+			serviceInstance.Status.LastOperation.State = "failed"
+			meta.SetStatusCondition(&serviceInstance.Status.Conditions, metav1.Condition{
+				Type:               korifiv1alpha1.UpdateFailedCondition,
```

Comment:
I guess that if the update failed with unrecoverable error, there is no point for the controller to retry update, correct?

If so, you could ammend the `isFailed` function to return try if the `UpdateFailedCondition` is true in order to prevent further update attempts with the broker